### PR TITLE
Evm speedup

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Broad reaching optimizations to speed up EVM calculations
* Generally speaking, use `int` and `long` where it is more appropriate than `UInt256` (memory indexes mostly)
* Move the internal stack to `Bytes` from `UInt256`
* Re-work the flow of many operations to account for the above

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by shemnon** [comment]
PR is built off of #2790, once that is merged the diffs will drop in half or more.

**Comment by atoulme** [comment]
Tag me for review whenever it’s time and happy to take a look!

**Comment by shemnon** [comment]
:fire: 20% median improvement, 130%+ worst case improvement :fire:
[Spreadsheet](https://docs.google.com/spreadsheets/d/1PG_VtetTLzYBriYsHy7hZuTcpKPsCTENGGmE-V5JuPc/edit#gid=1362220507)

**Comment by shemnon** [comment]
@atoulme it's only waiting on #2790 to merge, and will look like https://github.com/shemnon/besu/pull/10 once that happens.  If you want to :eyes: before the move merges you can start there.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java
Comment (atoulme) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java:237]: Is this expected?

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/handshake/ecies/ResponderHandshakeMessageV1.java
Comment (atoulme) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/handshake/ecies/ResponderHandshakeMessageV1.java:34]: Seems unrelated, but ok.

File: ethereum/rlp/build.gradle
Comment (atoulme) [inline ethereum/rlp/build.gradle:37]: Seems unrelated, but it's a good change.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java
Comment (shemnon) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java:237]: I had hard coded the wrong number - https://github.com/hyperledger/besu/blob/fd59570e2cb30425bcae5f231e4d22e9ef037410/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/IstanbulGasCalculator.java#L145

File: ethereum/rlp/build.gradle
Comment (shemnon) [reply ethereum/rlp/build.gradle:37]: It's related to the library transition, should have been in the prior PR.  This way the EVM lib does not drag in vertx.

File: evm/src/main/java/org/hyperledger/besu/evm/AccessListEntry.java
Comment (atoulme) [inline evm/src/main/java/org/hyperledger/besu/evm/AccessListEntry.java:26]: Is this just code simplification, or how does it help with speedup? Just curious.

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/handshake/ecies/ResponderHandshakeMessageV1.java
Comment (shemnon) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/handshake/ecies/ResponderHandshakeMessageV1.java:34]: Yea, some cleanups slipped in.  There will be a bigger Java 17 pr later.

File: evm/src/main/java/org/hyperledger/besu/evm/fluent/SimpleAccount.java
Comment (atoulme) [inline evm/src/main/java/org/hyperledger/besu/evm/fluent/SimpleAccount.java:119]: NGMI indeed

File: evm/src/main/java/org/hyperledger/besu/evm/AccessListEntry.java
Comment (shemnon) [reply evm/src/main/java/org/hyperledger/besu/evm/AccessListEntry.java:26]: I don't want to drag in JSON deserializers in the jar, which creates a runtime jackson dependency.  Annotations are discarded if the class isn't found so using build time only annotations preserves the jackson auto-serialization while not adding a large library dependency.

File: evm/src/main/java/org/hyperledger/besu/evm/internal/Words.java
Comment (atoulme) [inline evm/src/main/java/org/hyperledger/besu/evm/internal/Words.java:87]: Long.MAX_VALUE, not Integer.MAX_VALUE

File: evm/src/main/java/org/hyperledger/besu/evm/internal/Words.java
Comment (atoulme) [inline evm/src/main/java/org/hyperledger/besu/evm/internal/Words.java:90]: same comment here

File: evm/src/main/java/org/hyperledger/besu/evm/internal/Words.java
Comment (shemnon) [reply evm/src/main/java/org/hyperledger/besu/evm/internal/Words.java]: Done

File: evm/src/main/java/org/hyperledger/besu/evm/internal/Words.java
Comment (shemnon) [reply evm/src/main/java/org/hyperledger/besu/evm/internal/Words.java]: Done

File: evm/src/main/java/org/hyperledger/besu/evm/fluent/SimpleAccount.java
Comment (shemnon) [reply evm/src/main/java/org/hyperledger/besu/evm/fluent/SimpleAccount.java]: Changed to a more direct exception text for toy and simple.

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/handshake/ecies/ResponderHandshakeMessageV1.java
Comment (shemnon) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/handshake/ecies/ResponderHandshakeMessageV1.java]: undone.

**Review by atoulme — APPROVED** [review]
LGTM

**Pull request merged**
